source/common/http/utility.cc

```diff
@@ -562,5 +563,51 @@ void Utility::traversePerFilterConfigGeneric(
   }
 }
 
+std::string Utility::PercentEncoding::encode(absl::string_view value) {
+  std::string encoded;
+  for (size_t i = 0; i < value.size(); ++i) {
+    const char& ch = value[i];
+    // Unreserved characters. unreserved = ALPHA / DIGIT / "-" / "." / "_" / "~".
```

Comment:
per https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses this should be 0x20-0x7E except 0x25 (`%`), perhaps a grpc-message specific method is appropriate.

Reply:
:+1: